### PR TITLE
Updates login code to use mobile login headers and cookies

### PR DESCRIPTION
**Pull request by s992**
This allows us to grab the oAuth token when logging in. Once change to existing functionality is that `body.transfer_parameters` no longer exists, so we have to grab the Steam ID from `oAuth.steamid`.

If the user has logged in through other means (steam-user, etc.), they'll still have to get their oAuth token via the web chat.

Most of this comes from https://github.com/geel9/SteamAuth, just translated into Node.

### CODE REVIEW

**Comment by DoctorMcKay** [comment]
I don't like setting those headers on *every* request.

**Comment by s992** [comment]
They're probably only necessary on the initial login. Tomorrow I'll change them and test.

**Comment by s992** [comment]
Just pushed a commit that's only using the headers on the login requests. Everything still seems to be working fine. I also removed an extraneous cookie that I had added in my earlier commit.

File: index.js [hunk 1]
```diff
@@ -62,13 +83,17 @@ SteamCommunity.prototype.login = function(details, callback) {
 			"remember_login": "true",
 			"rsatimestamp": json.timestamp,
 			"twofactorcode": details.twoFactorCode || "",
-			"username": details.accountName
+			"username": details.accountName,
+			"oauth_client_id": "DE45CD61",
+			"oauth_scope": "read_profile write_profile read_client write_client",
+			"loginfriendlyname": "#login_emailauth_friendlyname_mobile"
```
Comment (DoctorMcKay) [inline index.js]: You have `loginfriendlyname` defined twice.

Comment (s992) [inline index.js]: Good catch, removed the empty one.

**Comment by Lagg** [comment]
I can confirm this works on top of current HEAD. Wish SteamUser could still be used though. Valve hates polling. Almost as much as they hate good security practices.

**Pull request merged**